### DISCUSSION TITLE
do i need to mount my ssh key into the container to use sftp or something?

### DISCUSSION THREAD

**Discussion by kinghat** [garethgeorge/backrest#245] [Q&A]
maybe im confused about how this is working but when using a container and trying to add an existing repo i get:

> Operation failed: [unknown] failed to init repo: init failed: command "/bin/restic-0.16.4 init --json -o sftp.args=-oBatchMode=yes" failed: exit status 1 Process STDOUT: subprocess ssh: ssh: Could not resolve hostname restic: Try again Fatal: create repository at sftp:restic:/path/to/existing/restic/repo failed: unable to start the sftp session, error: error receiving version packet from server: server unexpectedly closed connection: unexpected EOF

using the repo config:

```json
{
  "id": "nas",
  "uri": "sftp:restic:/path/to/existing/restic/repo",
  "password": "existing-repo-password",
  "prunePolicy": {
    "maxFrequencyDays": null,
    "maxUnusedPercent": null
  }
}
```

with an ssh config file on host:

```conf
Host restic
    Hostname nas.lan
    ServerAliveInterval 60
    ServerAliveCountMax 240
```

i can ssh from container host to `nas` just fine.

**Comment by garethgeorge** [comment]
Hey, if you're using a docker container yes you'll need to make sure your key file is in the container and that restic loads it using `sftp.args=-i <keyfile path>` I believe.

**Reply by kinghat** [reply]
it backrest setup to connect to existing repos?

> Operation failed: [unknown] failed to init repo: init failed: command "/bin/restic-0.16.4 init --json -o sftp.args=-i /data/id_ed25519_restic" failed: exit status 1 Process STDOUT: Fatal: the init command expects no arguments, only options - please see `restic help init` for usage and flags

**Reply by kinghat** [reply]
> it backrest setup to connect to existing repos?
> 
> > Operation failed: [unknown] failed to init repo: init failed: command "/bin/restic-0.16.4 init --json -o sftp.args=-i /data/id_ed25519_restic" failed: exit status 1 Process STDOUT: Fatal: the init command expects no arguments, only options - please see `restic help init` for usage and flags

ok i got it all working. couple things:

- i gave it an incorrect password as i was worried it would overwrite my existing repo(not sure that would have helped anyways).
- the repo was added but didnt tell me the password was incorrect nor after that its incorrect.
- you cant edit the password field after the repo is added and the generate button still exists.
- the password field isnt obfuscated.
